### PR TITLE
fix(FEC-10335): ad doesn't play correctly on same video tag with MSE

### PR DESCRIPTION
**Pull request by Yuvalke**
### Description of the Changes

IMA probably changed their adloaded timing which causes issues on our logic of playing ads with MSE.
issue: video detached 6-8 sec before midroll supposes to show up which causes an empty video tag.
Solution: detach the media when the ad break starts.

Solve FEC-10335.

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/player.js [hunk 1]
```diff
@@ -1811,13 +1811,15 @@ export default class Player extends FakeEventTarget {
         }
       });
       if (this.config.playback.playAdsWithMSE) {
-        this._eventManager.listen(this, AdEventType.AD_LOADED, (event: FakeEvent) => {
+        this._eventManager.listen(this, AdEventType.AD_LOADED, event => {
           if (event.payload.ad.linear) {
-            this._detachMediaSource();
+            this._eventManager.listenOnce(this, AdEventType.AD_BREAK_START, () => {
+              this._detachMediaSource();
+            });
```
Comment (OrenMe) [inline src/player.js:1818]: do we still need the AD_LOADED? why not register to AD_BREAK_START only?
If it is because of the ad.linear that only comes in AD_LOADED then I assume we can take the ad type from the current ad we have in the ad controller.

File: src/player.js [hunk 2]
```diff
@@ -1811,13 +1811,15 @@ export default class Player extends FakeEventTarget {
         }
       });
       if (this.config.playback.playAdsWithMSE) {
-        this._eventManager.listen(this, AdEventType.AD_LOADED, (event: FakeEvent) => {
+        this._eventManager.listen(this, AdEventType.AD_LOADED, event => {
```
Comment (OrenMe) [inline src/player.js:1814]: why did you remove the type?

File: src/player.js [hunk 3]
```diff
@@ -1811,13 +1811,15 @@ export default class Player extends FakeEventTarget {
...
           if (event.payload.ad.linear) {
-            this._detachMediaSource();
+            this._eventManager.listenOnce(this, AdEventType.AD_BREAK_START, () => {
+              this._detachMediaSource();
+            });
           }
         });
-        this._eventManager.listen(this, AdEventType.AD_BREAK_END, this._attachMediaSource);
-        this._eventManager.listen(this, AdEventType.AD_ERROR, this._attachMediaSource);
+        this._eventManager.listen(this, AdEventType.AD_BREAK_END, () => this._attachMediaSource());
```
Comment (OrenMe) [inline src/player.js:1821]: was this a mistake? the target is this so the context should remain the same in the handler

File: src/player.js (hunk 2 above)
Comment (Yuvalke) [reply src/player.js:1814]: Mistake, I'll add it

File: src/player.js (hunk 3 above)
Comment (Yuvalke) [reply src/player.js:1821]: both correct wanted to be more explicit.
